src/binaries.ts

```diff
@@ -0,0 +1,80 @@
+import { pipe } from "fp-ts/lib/function";
+import * as E from "fp-ts/lib/Either";
+import * as t from "io-ts";
+
+/**
+Check wether the input is a valid Uint8Array
+ */
+// eslint-disable-next-line
+const checkUint8Array = (toCheck: any): boolean =>
+  Array.isArray(toCheck) &&
+  toCheck.length > 0 &&
+  // eslint-disable-next-line
+  toCheck.every((e: any) => Number.isInteger(e) && e >= 0 && e <= 255);
+
+/**
+In order to be a valid array to generate a buffer:
+1. It must be a non empty array
+2. Every element must be an integer with the value between 0 and 255
+If an element of the array is outside of this range, it will be converted with the % operator
+from the Buffer.from() method, in order to avoid reading problems this function will consider any Array with at
+least one number outside of the range 0 - 255 non valid.
+ */
+
+const Uint8ArrayValidator = (
+  arr: any, // eslint-disable-line
+  c: t.Context
+): E.Either<t.Errors, Buffer> =>
+  pipe(
+    arr,
+    E.fromPredicate(
+      () => checkUint8Array(arr),
+      () => [{ context: c, value: "Invalid Uint8Array" }]
+    ),
+    E.map(a => Buffer.from(a))
+  );
+
+/**
+Get a Buffer from a valid number[]  
+ */
+
+export const BinaryFromArray = new t.Type<Buffer, unknown, unknown>(
+  "BinaryFromArray",
+  (input): input is Buffer =>
+    Buffer.isBuffer(input) && input.length ? true : false,
```

Comment:
`input.length ? true : false`
seems to be tautological: remove the ternary operator

Reply:
why do you want to exclude empty values?

Reply:
@fabriziopapi you are right, @gunzip  I was thinking about this type `Binary` for situations such as downloading a file and so I thought that an empty `Buffer` could be an error case... but maybe I am wrong here.

Reply:
I've update the gist to use identity for `encoding` in case that's what you want to.
My concern with the current version are:
1. methods that return booleans hould be typeguards instead
2. methods that return booleans must be prefixed with `is` or `has` 
3. poor naming: `Uint8ArrayValidator` convert the input to a buffer, while the name suggest it's a read only op
4. missing usage of `failure` / `success` methods (ie. hardcoding `[{ context: c, value: "Invalid Uint8Array" }]`)
5. mismatch between `is` and `decode` methods (ie. `BinaryFromArray.is ` doesn't check that the buffer is not empty, while decode does)